pkg/listwatch/listwatch.go

```diff
@@ -262,6 +262,9 @@ func (mlw *multiListerWatcher) newMultiWatch(resourceVersions map[string]string,
 		o.ResourceVersion = resourceVersions[n]
 		w, err := lws.lw.Watch(*o)
 		if err != nil {
+			mlw.lock.Unlock()
+			mlw.Stop()
+			mlw.lock.Lock()
```

Comment:
Same comment as in #49 (don't unlock; call StopInternal() instead)

Reply:
Applied the same change. Thanks!